.github/workflows/extended.yml

```
@@ -33,16 +33,46 @@ on:
   push:
     branches:
       - main
+  issue_comment:
+    types: [created]
+
+permissions:
+  pull-requests: write
 
 jobs:
+  # Check issue comment and notify that extended tests are running
+  check_issue_comment:
+    name: Check issue comment
+    runs-on: ubuntu-latest
+    if: github.event.issue.pull_request && github.event.comment.body == 'run extended tests'
+    steps:
+      - uses: actions/github-script@v7
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            github.rest.issues.createComment({
+              issue_number: context.issue.number,
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              body: "Running extended tests..."
+            })
+
   # Check crate compiles and base cargo check passes
   linux-build-lib:
     name: linux build test
     runs-on: ubuntu-latest
     container:
       image: amd64/rust
+    if: |
+      github.event_name == 'push' ||
+      (github.event_name == 'issue_comment' && github.event.issue.pull_request && github.event.comment.body == 'run extended tests')
     steps:
       - uses: actions/checkout@v4
+        with:
+          # Check out the pull request branch if triggered by a comment
+          ref: ${{ github.event_name == 'issue_comment' && github.event.issue.pull_request.head.ref || github.ref }}
```

A fully formatted ref should work. Probably there are other ways to fetch the same info as well. 

```suggestion
          ref: ${{ github.event_name == 'issue_comment' && format('refs/pull/{0}/head', github.event.issue.number) || github.ref }}
```

[Test PR](https://github.com/danila-b/datafusion/pull/3)
[Main branch workflow state](https://github.com/danila-b/datafusion/blob/main/.github/workflows/extended.yml#L74)
[Checkout commit matches PR head](https://github.com/danila-b/datafusion/actions/runs/13473711396/job/37650297887)

---

🤔  
- https://github.com/danila-b/datafusion/pull/3 doesn't seem to have run the extended tests

https://github.com/danila-b/datafusion/pull/3#issuecomment-2697904032

---

Apparently it did run them, it's just not visible in GitHub UI. I have just double checked it with a bit different implementation on the same branch. 

Unfortunately, it seems that `issue_comment` event trigger is associated with default branch, and hence it does not add anything to the checks or UI of the PR itself. I can take a look if there is a better way to launch check associated with PR during the weekend, but from the top of my head, there are a couple of options:

1. Use `workflow_dispatch` event and trigger extended tests for the branch from the GitHub actions UI tab instead of doing so by comment on the PR
2. Use [3rd party action](https://github.com/myrotvorets/set-commit-status-action) to set check statuses when the run is triggered by a comment
3. Maybe [slash command 3rd party action](https://github.com/marketplace/actions/slash-command-dispatch) can also work for triggering commands a bit better since it uses dispatch event under the hood as well

---

I tried messing around with this again this morning and didn't have much luck

---

Thank you again for the help -- I am out of my league in terms of github actions.

---

@alamb I gave it a stab implementing this here: https://github.com/apache/datafusion/pull/15101
It should dispatch extended tests on a PR comment, and also create and update the check marker on the PR. 

Testing this is a bit of a pain since I don't know any better way than to just merge the workflow to the main branch of the fork and run several test PRs. At least seems to work properly from a quick glance, will test a bit more during the weekend.